### DISCUSSION TITLE
Can i get some help?

### DISCUSSION THREAD

**Discussion by havaka** [andsfonseca/dota2-rpc-client#45] [Q&A]
I'm noob at programming, and i cant understand how to change discord rpc text that appears in my status. Can someone explain me pls what file should i change?
By the way, I'm talking not about on/off some settings, I'm asking how to completely change all text in my status

**Comment by andsfonseca** [comment]
Can you tell me which text u want to change?
I can check here if needed.

![image](https://user-images.githubusercontent.com/7833466/200430382-4acf3f01-cbc3-4ac6-9952-04d7474b7b20.png)

Red - Image Description (Networth)
Green - Title (Ex. Playing with Bots)
Blue - Details (Ex. Testing a Hero, or KDA)
White - Time
Orange - Small Image Description (Aghanim, Smoke, Roshan)

**Reply by havaka** [reply]
I wanna change everything in my discord activity. I just thought u wrote all of it as one file, and I'd like to know its destination

**Reply by andsfonseca** [reply]
In my code I've encapsulated the Discord SDK inside a service.
However, it uses this structure provided by the SDK.

This is a [Generic Example](https://discord.com/developers/docs/game-sdk/activities#updateactivity-example) but you can see something similar running inside the InterpretJson function in [DotaService.cpp](https://github.com/andsfonseca/dota2-rpc-client/blob/master/src/services/DotaService.cpp)

https://github.com/andsfonseca/dota2-rpc-client/blob/d1db54d58578263805bc8fcb20e2318961ec7022/src/services/DotaService.cpp#L495-L526
